src/java.base/share/classes/java/lang/annotation/ElementType.java

```diff
@@ -26,106 +26,213 @@
 package java.lang.annotation;
 
 /**
- * The constants of this enumerated class provide a simple classification of the
- * syntactic locations where annotations may appear in a Java program. These
- * constants are used in {@link java.lang.annotation.Target Target}
- * meta-annotations to specify where it is legal to write annotations of a
- * given type.
+ * A syntactic location where an annotation may appear in Java code.
+ * An annotation interface may optionally restrict its usage to a
+ * particular subset of these locations using the {@link
+ * java.lang.annotation.Target @Target} meta-annotation.
  *
- * <p>The syntactic locations where annotations may appear are split into
- * <em>declaration contexts</em>, where annotations apply to declarations, and
- * <em>type contexts</em>, where annotations apply to types used in
- * declarations and expressions.
+ * <p>For example, an annotation of the following type may only appear
+ * as part of a type parameter or local variable declaration:
  *
- * <p>The constants {@link #ANNOTATION_TYPE}, {@link #CONSTRUCTOR}, {@link
- * #FIELD}, {@link #LOCAL_VARIABLE}, {@link #METHOD}, {@link #PACKAGE}, {@link
- * #MODULE}, {@link #PARAMETER}, {@link #TYPE}, and {@link #TYPE_PARAMETER}
- * correspond to the declaration contexts in JLS {@jls 9.6.4.1}.
+ * {@snippet id='example' :
+ * @Target({ElementType.TYPE_PARAMETER, ElementType.LOCAL_VARIABLE})
+ * public @interface MyAnnotation {}
+ * }
  *
- * <p>For example, an annotation whose interface is meta-annotated with
- * {@code @Target(ElementType.FIELD)} may only be written as a modifier for a
- * field declaration.
+ * <h2>Declaration annotations and type-use annotations</h2>
```

Comment:
Meta: if we add an explicit id to the h2 tag, people can have a stable link to this subsection instead of having to update their link for each release, as the javadoc tool automatically generates an unstable id.

Reply:
Good idea. I guess we should always do this.